frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -65,9 +76,144 @@ function patientAliasFilterWhereVariables(value: string): PatientAliasWhere[] {
+  title: "Access unauthorized",
+  content:
+    "You are not authorized to access PHI data. If you would like to request access, please reach out to the administrator.",
+};
+
+export default function PatientsPage({
+  setUserEmail,
+}: {
+  setUserEmail: (userEmail: string | null) => void;
+}) {
   const params = useParams();
 
+  const [searchVal, setSearchVal] = useState<string[]>([]);
+  const [inputVal, setInputVal] = useState("");
+  const [showDownloadModal, setShowDownloadModal] = useState(false);
+
+  const [phiEnabled, setPhiEnabled] = useState(false);
+  const [patientIdsTriplets, setPatientIdsTriplets] = useState<
+    PatientIdsTriplet[]
+  >([]);
+  const [alertModal, setAlertModal] = useState<{
+    show: boolean;
+    title: string;
+    content: string;
+  }>({ show: false, title: "", content: "" });
+
+  async function fetchPatientIdsTriplets(
+    patientIds: string[]
+  ): Promise<string[]> {
+    try {
+      const response = await fetch("https://localhost:4000/mrn-search", {
+        method: "POST",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(patientIds),
+      });
+
+      if (response.status === 403) {
+        setAlertModal({
+          show: true,
+          ...unauthorizedWarning,
+        });
+        return [];
+      }
+
+      if (response.status === 401) {
+        const width = 800;
+        const height = 800;
+        const left = (window.screen.width - width) / 2;
+        const top = (window.screen.height - height) / 2;
+
+        window.open(
+          "https://localhost:4000/login",
+          "_blank",
+          `toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width=${width}, height=${height}, top=${top}, left=${left}`
+        );
+        return [];
+      }
+
+      const data: PatientIdsTriplet[] = await response.json();
+      setPatientIdsTriplets(data);
+      return data.map((d) => addCDashToCMOId(d.cmoId));
+    } catch (error) {
+      console.error(error);
+      return [];
+    }
+  }
+
+  const handleSearch = async () => {
+    const uniqueQueries = parseSearchQueries(inputVal);
+    if (phiEnabled) {
+      const newQueries = await fetchPatientIdsTriplets(uniqueQueries);
+      if (newQueries.length > 0) setSearchVal(newQueries);
+    } else {
+      setSearchVal(uniqueQueries);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("message", handleLogin);
+
+    function handleLogin(event: any) {
+      if (event.origin !== "https://localhost:4000") return;
+      setUserEmail(event.data);
+      setAlertModal({
+        show: true,
+        ...phiWarning,
+      });
+      handleSearch();
+    }
+
+    return () => {
+      window.removeEventListener("message", handleLogin);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [phiEnabled]);
+
+  PatientsListColumns = useMemo(() => {
```

Comment:
this seems very odd.  you are using PatientsListColumns inside of the useMemo which is mean to create it?  
Also, a varaible like PatientsListColumns should always have a var, let or const before it.   and it should be camelcase.  capitalized is should be reserved for Reach component.

Reply:
just to be clear, useMemo should definately not refer to itself.  it should be a pure function that computes something from its dependencies (or some known static value)

Reply:
Let's discuss this one live. I have a few clarifying questions. Thanks, Aaron.